clap_complete/src/generator/utils.rs

```diff
@@ -47,8 +47,15 @@ pub fn subcommands(p: &Command) -> Vec<(String, String)> {
             sc.get_name(),
             sc_bin_name
         );
-
         subcmds.push((sc.get_name().to_string(), sc_bin_name.to_string()));
+
+        for alias in sc.get_visible_aliases() {
+            debug!(
+                "subcommands:iter: alias={}, bin_name={}",
+                alias, sc_bin_name
+            );
+            subcmds.push((alias.to_string(), sc_bin_name.to_string()));
```

Comment:
This only affects zsh and bash.  We need to cover the other shells we support.

Reply:
I only edited zsh and bash generators, because only they supported visible_alias at all (by listing aliased subcommands, just not their subcommand).

I just wrote commits that are adding support for visible_alias to fish, elvish and powershell. I have limited access to computers this week, so I cannot test them, but the snapshot tests are looking good.